File: day-01/solution.kt
//INCLUDE ../commons/io.kt

import aoc.commons.readLines


fun main() {
    val lines = readLines("day-01/input.txt")
    val nums = lines.map { it.toInt() }
    val threesums = Array(nums.size - 2) { 0 }
    for (i in 0 until threesums.size) {
        threesums[i] = nums[i] + nums[i + 1] + nums[i + 2]
    }
    var count = 0
    for (i in 1 until threesums.size) {
        if (threesums[i] > threesums[i - 1]) {
            count++
        }
    }
    println(count)
}

File: day-01/solution.ts
import {readLines} from "../commons/io";

function main() {
    const lines = readLines("day-01/input.txt");
    const nums = lines.map(Number);
    const threesums = new Array(nums.length - 1);
    for (let i = 0; i < threesums.length; i++) {
        threesums[i] = nums[i] + nums[i + 1] + nums[i + 2];
    }
    let count = 0;
    for (let i = 1; i <= threesums.length; i++) {
        if (threesums[i] > threesums[i - 1]) {
            count++;
        }
    }
    console.log(count);
}

main()